framework/include/controls/PIDTransientControl.h

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// moose includes
+#include "Control.h"
+
+/**
+ * A time-dependent control of an input parameter or a postprocessor, which aims at
+ * making a postprocessor match a desired value.
+ */
+class PIDTransientControl : public Control
+{
+public:
+  /**
+   * Class constructor
+   * @param parameters Input parameters for this Control object
+   */
+  static InputParameters validParams();
+
+  PIDTransientControl(const InputParameters & parameters);
+
+  virtual void execute() override;
+
+private:
+  /// The current value of the target postprocessor
+  const PostprocessorValue & _current;
+  /// The target 1D time-dependent function for the postprocessor
+  const Function & _target;
+  /// Integral of the error
+  Real _integral;
+  /// The coefficient multiplying the integral of the error
+  const Real _Kint;
+  /// The coefficient multiplying the error
+  const Real _Kpro;
+  /// The coefficient multiplying the derivative of the error
+  const Real _Kder;
+  /// The time to start the PID controller on
+  const Real _start_time;
+  /// The time to stop using the PID controller on
+  const Real _stop_time;
```

Comment:
Should not enabling/disabling of the object be done via the `TimePeriod` class? Or we are not able to enable/disable `Control`-derived objects? Just asking...

Reply:
I get this:
*** ERROR ***
The following error occurred in the object "diff", of type "TimePeriod".

The desired parameter 'Controls/integral_value/enable' was not located for the 'diff' object, it either does not exist or has not been declared as controllable.

Reply:
Ok, thanks for try this. This may not be possible on purpose. I just wanted to be sure.